src/App.jsx

```diff
@@ -14,45 +16,106 @@ export const goodsFromServer = [
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [sortedGoods, setSortedGoods] = useState([...goodsFromServer]);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(null);
+
+  const sortAlphabetically = () => {
+    if (!isReversed || (isReversed && sortType !== 'alphabetical')) {
+      const orderAbc = isReversed
+        ? [...sortedGoods].sort().reverse()
+        : [...sortedGoods].sort();
+
+      setSortedGoods(orderAbc);
+      setSortType('alphabetical');
+    }
+  };
+
+  const sortByLength = () => {
+    if (!isReversed || (isReversed && sortType !== 'length')) {
+      const orderLength = isReversed
+        ? [...sortedGoods].sort((a, b) => b.length - a.length)
+        : [...sortedGoods].sort((a, b) => a.length - b.length);
+
+      setSortedGoods(orderLength);
+      setSortType('length');
+    }
+  };
+
+  const reverseOrder = () => {
+    const reverOrder = [...sortedGoods].reverse();
+
+    setSortedGoods(reverOrder);
+    setIsReversed(!isReversed);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const resetOrder = () => {
+    setSortedGoods([...goodsFromServer]);
+    setIsReversed(false);
+    setSortType(null);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames(
+            'button',
+            'is-info',
+            { 'is-light': sortType !== 'alphabetical' },
+          )}
+          onClick={sortAlphabetically}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={
+            classNames(
+              'button',
+              'is-success',
+              { 'is-light': sortType !== 'length' },
+            )}
+          onClick={sortByLength}
+
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={
+            classNames(
+              'button',
+              'is-warning',
+              { 'is-light': isReversed === false },
+            )}
+          onClick={reverseOrder}
+        >
+          Reverse
+        </button>
+
+        {sortedGoods.join('') !== goodsFromServer.join('') && (
```

Comment:
It is better to transfer this condition to a separate variable for better readability and maintainability of the code